contracts/HubPool.sol

```diff
@@ -144,67 +158,73 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         emit BondSet(address(newBondToken), newBondAmount);
     }
 
+    function setRefundProposalLiveness(uint64 newRefundProposalLiveness) public onlyOwner {
+        refundProposalLiveness = newRefundProposalLiveness;
+        emit RefundProposalLivenessSet(newRefundProposalLiveness);
+    }
+
+    function setIdentifier(bytes32 newIdentifier) public onlyOwner {
+        identifier = newIdentifier;
+        emit IdentifierSet(newIdentifier);
+    }
+
     function setCrossChainContracts(
-        uint256 chainId,
-        AdapterInterface adapter,
+        uint256 l2ChainId,
+        address adapter,
         address spokePool
     ) public onlyOwner noActiveRequests {
-        require(address(crossChainContracts[chainId].adapter) == address(0), "Contract already set");
-        crossChainContracts[chainId] = CrossChainContract(adapter, spokePool);
+        require(address(crossChainContracts[l2ChainId].adapter) == address(0), "Contract already set");
+        crossChainContracts[l2ChainId] = CrossChainContract(AdapterInterface(adapter), spokePool);
+        emit CrossChainContractsSet(l2ChainId, adapter, spokePool);
     }
 
     /**
      * @notice Whitelist an origin token <-> destination token route.
      */
     function whitelistRoute(
+        uint256 destinationChainId,
         address originToken,
-        address destinationToken,
-        uint256 destinationChainId
+        address destinationToken
     ) public onlyOwner {
         whitelistedRoutes[originToken][destinationChainId] = destinationToken;
```

Comment:
should this function also call `setEnableRoute` cross-chain on the spoke pool? No need to do anything now but maybe worth adding a TODO comment to remind ourselves to implement in a separate PR

Reply:
yes! in the future, it will. will add a TODO.